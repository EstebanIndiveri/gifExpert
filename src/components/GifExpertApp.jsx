import React, { Fragment } from 'react';
import { useState } from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';
const GifExpertApp = ({defaultCategories=[]}) => {
    // const categories=['One Punch','Samurai X','Dragon Ball']

    const [categories,setCategories]=useState(defaultCategories);


    return ( 
        <Fragment>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr/>
        <ul>
            {categories.map(categorie=>(
                <GifGrid 
                key={categorie}
                categorie={categorie}/>
            ))}
        </ul>
        </Fragment>
     );
}
 
export default GifExpertApp;