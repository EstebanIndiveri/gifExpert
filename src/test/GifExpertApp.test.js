import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import GifExpertApp from '../components/GifExpertApp';
describe('match con el snap', () => {
    test('GifExpertApp/>', () => {
        const wrapper=shallow(<GifExpertApp/>)
        expect(wrapper).toMatchSnapshot();
    });
    test('muestra lista de categories', () => {
        const categories=['one punch','dragon ball'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
})
