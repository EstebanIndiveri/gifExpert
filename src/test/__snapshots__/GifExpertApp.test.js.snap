// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match con el snap GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`match con el snap muestra lista de categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      categorie="one punch"
      key="one punch"
    />
    <GifGrid
      categorie="dragon ball"
      key="dragon ball"
    />
  </ul>
</Fragment>
`;
